storage/handler.go

```diff
@@ -87,15 +87,21 @@ func ListByQuery(l *ListRequest, m map[string][]string) {
 	case *Roles:
 		var res Roles
 		for _, role := range *val {
-			res = append(res, *role.withMembers(m["member"]).withIDs(m["id"]))
+			filteredRole := role.withMembers(m["members"]).withIDs(m["id"])
```

Comment:
A note on the filter: If someone passed unknown url query param to roles or policies list API then that isn't handled here. All values will be returned unless there is some parameter validation happening before filter is called.

Reply:
Also, not sure if query params should be singular or plural. I added support for plural i.e. "members" not "member"

Reply:
can you clarify this? @aeneasr

Reply:
Let's keep it singular. Plural can be both AND and OR and we don't want to deal with that right now.

Regarding unknown parameters - that's fine. If it's unknown, it will be ignored.

Reply:
Okay, those were my concerns as well but we dont allow passing multiple values for let's say actions then the problem of calling list api multiple times is still there. The way I have written it if someone calls `../policies/list?actions="delete"&action="create"&subject="122"`, the filter will still work.

Reply:
@aeneasr